pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -457,7 +451,7 @@ func updateConfigSpec(
 	UpdateHardwareConfigSpec(config, configSpec, &vmClassSpec)
 	UpdateConfigSpecCPUAllocation(config, configSpec, &vmClassSpec, updateArgs.MinCPUFreq)
 	UpdateConfigSpecMemoryAllocation(config, configSpec, &vmClassSpec)
-	UpdateConfigSpecExtraConfig(config, configSpec, updateArgs.ConfigSpec, updateArgs.VMImageStatus, &vmClassSpec, vmCtx.VM, updateArgs.ExtraConfig)
+	UpdateConfigSpecExtraConfig(config, configSpec, updateArgs.ConfigSpec, &vmClassSpec, vmCtx.VM, updateArgs.ExtraConfig)
 	UpdateConfigSpecChangeBlockTracking(config, configSpec, updateArgs.ConfigSpec, vmCtx.VM.Spec)
```

Comment:
Not your change, but should this be `vmClassSpec`?  It sound like the [function that populates ](https://github.com/vmware-tanzu/vm-operator/pull/36/files#diff-d5b7c2c460dc9978a628f2cb03390868152b8d5fdd4e2dd99c537ec36a2f6143R864)the `updateArgs.ConfigSpec` (`vmUpdateGetArgs`) doesn't take CBT into account, so IIUC, CBT will never get set?

Reply:
if `vmClassConfigSpec` [is not nil](https://github.com/vmware-tanzu/vm-operator/blob/a96f5cf1732685251cd1d377c80fba5468587dc8/pkg/vmprovider/providers/vsphere/virtualmachine/configspec.go#L27), it will contain in the `updateArgs.ConfigSpec` . Users can set CBT in the VM Class configSpec I think?

Reply:
Yes, I do think the existing behavior is suspect - the VM Spec should take priority when set. This is something we can sort out as we flesh out more of the ConfigSpec CL Deploy work (likely, the class value will only be used at create time, again only if the VM Spec is not set).